bamboo_twig_i18n/src/TwigExtension/I18n.php

```diff
@@ -85,4 +87,33 @@ public function formatDate(TwigEnvironment $env, $date, $type = 'medium', $forma
     return NULL;
   }
 
+  /**
+   * Gets a translation of the entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity for which to get the translation.
+   * @param string|null $langcode
+   *   (optional) The language code of the translation to get. NULL (default) means to use
+   *   the user interface language for the page.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface|null
+   *   An entity object when translations exists or NULL.
+   */
+  public function getTranslation(EntityInterface $entity, $langcode = NULL) {
+    /** @var \Drupal\Core\Language\LanguageManagerInterface $language_manager */
+    $language_manager = $this->getLanguageManager();
+
+    if (!$langcode) {
+      $langcode = $this->getCurrentLanguage();
+    }
+
+    // Check the given langcode exists.
+    $languages = $language_manager->getLanguages();
+    if (!isset($languages[$langcode])) {
+      return NULL;
+    }
+
+    // Return the translation of this entity or NULL when not translated.
+    return $entity->hasTranslation($langcode) ? $entity->getTranslation($langcode) : NULL;
```

Comment:
So we agree to return `NULL` if the translation didn't exist ?

```twig
{{ entity|bamboo_i18n_get_translation('fr').title.value }}
```
will raise an error if `fr` translation didn't exist. It's fine for me and certainly a better API.

@tonifisler OK with this ?

Reply:
We need to return `NULL`  to cover #17